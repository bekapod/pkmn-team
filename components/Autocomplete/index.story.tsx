// tslint:disable-next-line:no-implicit-dependencies
import { storiesOf } from "@storybook/react";
import React from "react";
import Autocomplete, { AutocompleteDropdown } from ".";

storiesOf("Autocomplete", module).add("default", () => (
  <Autocomplete>
    <AutocompleteDropdown>
      <div>Item 1</div>
      <div>Item 2</div>
    </AutocompleteDropdown>
  </Autocomplete>
));
