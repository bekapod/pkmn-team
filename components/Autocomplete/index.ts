import styled from "styled-components/macro";
import * as variables from "../../helpers/variables";

export const AutocompleteDropdown = styled.div`
  flex: 1;
  width: 100%;
  background-color: ${variables.colors.white};

  > * {
    width: 100% !important;
  }
`;

const Autocomplete = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
`;

export default Autocomplete;
