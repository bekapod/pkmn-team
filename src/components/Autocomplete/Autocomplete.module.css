.container {
  display: flex;
  flex-direction: column;
  height: 100%;
}

.dropdown {
  flex: 1 0 auto;
  width: 100%;
  background-color: var(--color-white);

  & > * {
    width: 100% !important;
  }
}
