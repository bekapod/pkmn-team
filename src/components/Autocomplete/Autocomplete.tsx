import type { ComponentPropsWithRef, FunctionComponent } from 'react';
import classNames from 'classnames';

export const Autocomplete: FunctionComponent<ComponentPropsWithRef<'div'>> = ({
  className,
  ...props
}) => (
  <div
    className={classNames('flex', 'flex-col', 'h-full', className)}
    {...props}
  />
);

export const AutocompleteDropdown: FunctionComponent<
  ComponentPropsWithRef<'div'>
> = ({ className, ...props }) => (
  <div
    className={classNames(
      'flex-grow',
      'flex-shrink-0',
      'w-full',
      'bg-white',
      'children:w-full',
      className
    )}
    {...props}
  />
);
