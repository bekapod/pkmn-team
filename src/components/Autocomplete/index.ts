import styled from "styled-components/macro";
import * as variables from "../../helpers/variables";

export const AutocompleteDropdown = styled.div`
  flex: 1;
  width: 100%;
  padding: calc(${variables.spacing.sm}px / 2) 0;
  overflow-y: scroll;
  background-color: ${variables.colors.white};
`;

const Autocomplete = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
`;

export default Autocomplete;
