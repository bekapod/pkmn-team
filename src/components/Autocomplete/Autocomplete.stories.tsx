import { Meta } from '@storybook/react/types-6-0';
import { Autocomplete, AutocompleteDropdown } from './Autocomplete';

export default {
  title: 'Components/Autocomplete',
  component: Autocomplete
} as Meta;

export const Standard = (): JSX.Element => (
  <Autocomplete>
    <AutocompleteDropdown>
      <div>Item 1</div>
      <div>Item 2</div>
    </AutocompleteDropdown>
  </Autocomplete>
);
