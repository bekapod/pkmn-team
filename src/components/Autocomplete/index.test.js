// @flow
import React from "react";
import renderer from "react-test-renderer";
import Autocomplete, { AutocompleteDropdown } from ".";

describe("<Autocomplete />", () => {
  it("renders without crashing", () => {
    const tree = renderer.create(
      <Autocomplete>
        <AutocompleteDropdown>
          <span>Item 1</span>
          <span>Item 2</span>
        </AutocompleteDropdown>
      </Autocomplete>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
