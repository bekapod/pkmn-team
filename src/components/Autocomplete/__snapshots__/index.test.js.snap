// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Autocomplete /> renders without crashing 1`] = `
<div
  className="Autocomplete-sc-18qws42-1 jEZDkd"
>
  <div
    className="Autocomplete__AutocompleteDropdown-sc-18qws42-0 exSuOy"
  >
    <span>
      Item 1
    </span>
    <span>
      Item 2
    </span>
  </div>
</div>
`;
